python/Ice/filesystem/client/main.py

```diff
@@ -27,26 +27,31 @@ async def list_recursive(directory, depth=0):
     contents = await directory.listAsync()
 
     for node in contents:
-        assert node is not None  # The node proxies returned by list() are never null.
+        # The node proxies returned by list() are never null.
+        assert node is not None
 
+        # Check if this node is a directory by asking the remote object.
         subdir = await DirectoryPrx.checkedCastAsync(node)
-        kind = "directory" if subdir is not None else "file"
+
+        # We assume it's a file if it's not a directory.
+        kind = "(directory)" if subdir is not None else "(file)"
```

Comment:
A few languages were missing the parenthesis around `directory` and `file`. Python was one of them.